src/system.h

```diff
@@ -29,6 +29,12 @@
 # define strsignal(p) ""
 #endif
 
+#if _____LP64_____
```

Comment:
Pointer size is a poor proxy for the amount of available RAM. We should not let the hash table size depend on this.

Reply:
I disagree. The only people who will argue for the smaller hash table are those in the 16-32MB ram range. It's a perfectly reasonable assumption that:
a) No one will be running 16-32MB ram on 32bit; and
b) That if you want performance you should be building on 64bit (e.g 32bit userspace calling 64bit kernelspace is very slow)
c) There are few use cases for purposefully running a 32bit kernel on a 64bit machine

If you want to autoconf it, you can. As I said earlier - I'm not touching autoconf.

Reply:
How much ram does a hash table of size 0x1000 use? We do have a lot of user who use tinc on their router.

Reply:
I think these sizes are reasonable, but do we have to do this in compile time? Could we do it at run time? Sorry for sounding totally clueless.

Reply:
From @gsliepen 
> If we want to go this way, then instead of having a `hash_alloc_ ## t()` that returns a generic `hash_t *`, I'd rather have a `hash_alloc_define(t)` create a specialized `hash_ ## t ## _t`, that has a member variable `t keys[0x100]`, and declares that hash table as a static global variable. That way we also avoid malloc calls and pointer dereferencing as much as possible.

I tend to agree. Theres few reasons the hash table used for route cache should need use configuration (other than possibly at compile time).


Open addressing + size  increase  = more than enough room to spare :)

Reply:
I agree with @gsliepen's assessment.

Reply:
Maring as resolved as dealt with in later conversation.